src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -139,61 +121,36 @@ def update_stats(self, area, simulation_status, progress_info):
             self.area_market_stocks_stats.update(area)
 
-    def _send_results_to_areas(self, area):
-        stats = {
-            "kpi": self.kpi.performance_indices_redis.get(area.uuid, None)
-        }
-        area.endpoint_stats.update(stats)
-
     def update_area_aggregated_stats(self, area):
-        self._update_area_stats(area)
-        self._send_results_to_areas(area)
+        self._update_bids_offer_trades(area)
+        self._merge_cumulative_bills_into_bills_for_market_info(area)
         for child in area.children:
             self.update_area_aggregated_stats(child)
 
-    def _update_area_stats(self, area):
+    def _update_bids_offer_trades(self, area):
         if area.current_market is not None:
             self.bids_offers_trades[area.uuid] = area.current_market.get_bids_offers_trades()
-        bills = self.market_bills.bills_redis_results[area.uuid]
-        bills.update({
-            "penalty_cost": self.cumulative_bills.cumulative_bills_results[area.uuid]["penalties"],
-            "penalty_energy":
-                self.cumulative_bills.cumulative_bills_results[area.uuid]["penalty_energy"]})
-        area.stats.update_aggregated_stats({"bills": bills})
-
-
-class CumulativeGridTrades:
-    def __init__(self):
-        self.current_trades = {}
-        self.current_trades_redis = {}
-        self.current_balancing_trades = {}
-        self.accumulated_trades = {}
-        self.accumulated_trades_redis = {}
-        self.accumulated_balancing_trades = {}
-
-    def update(self, area):
-        market_type = \
-            "past_markets" if ConstSettings.GeneralSettings.KEEP_PAST_MARKETS else "current_market"
-        balancing_market_type = "past_balancing_markets" \
-            if ConstSettings.GeneralSettings.KEEP_PAST_MARKETS \
-            else "current_balancing_market"
-
-        if ConstSettings.GeneralSettings.KEEP_PAST_MARKETS:
-            self.accumulated_trades = {}
-            self.accumulated_trades_redis = {}
-            self.accumulated_balancing_trades = {}
-
-        self.accumulated_trades_redis, self.current_trades_redis = \
-            export_cumulative_grid_trades_redis(area, self.accumulated_trades_redis,
-                                                market_type)
-        self.accumulated_trades, self.current_trades = \
-            export_cumulative_grid_trades(area, self.accumulated_trades,
-                                          market_type, all_devices=True)
-        self.accumulated_balancing_trades, self.current_balancing_trades = \
-            export_cumulative_grid_trades(area, self.accumulated_balancing_trades,
-                                          balancing_market_type)
+            # TODO: (D3ASIM-2670) Trades are already stored in bids_offers_trades:
+            #  accumulation of area.stats.market_trades endpoint should be deprecated
+            self.last_energy_trades_high_resolution[area.uuid] = area.stats.market_trades
```

Comment:
`area.stats.market_trades` object is storing a dict of market time slots -> trades that happened on this market slot. Therefore in this case, the last_energy_trades_high_resolution is always storing the trades for the whole simulation. This is a potential memory issue, because this can cause the memory to grow in size indefinitely. 
If I have understood something wrong in the PR please ping me, because there are probably other uses of market_trades that I do not see in the PR.

Reply:
I agree! I started dealing with this, but realised, that it is more complex and created the D3ASIM-2670 story that deals with deprecating `area.stats.market_trades`.